degica/komoju-woocommerce
# Add shipping_address to the API request when the default shipping_address is the billing_address

# Context
https://github.com/degica/komoju-woocommerce/assets/15086337/642f6b48-847c-4c00-b1ae-8e9d8b914916

Paidy payments done via Woocommerce for physical_goods MID are not working. We are getting error 422 from Paidy with "payment_data[shipping_address]" is missing. This means that all physical goods MID merchants using Paidy for payments via Woocommerce will have failed payments.

This PR adds the `shipping_address` to the API request when the default `shipping_address` is the `billing_address

Fixes: https://app.asana.com/0/1203431131393579/1205964261339053/f

# Test instructions
**Setup**
- Follow [this](https://www.notion.so/WooCommerce-Testing-Demo-9bbada14ee314a379e141f8d512dde4f?pvs=4#944442c2cc364f7ca9ababde5ec81078) to setup local Woocommerce.
- Configure Woocommerce Komoju settings to connect to degica-mart of a local instance of HATS by setting the `KOMOJU Endpoint`.

<img width="1095" alt="Screen Shot 2024-01-18 at 14 49 35" src="https://github.com/degica/komoju-woocommerce/assets/15086337/119332ed-3d9e-4979-b5fd-3be223f8d95b">

- Configure degica-mart to use Paidy 1 gateway.
- Configure Paidy 1 to be a Gateway Type of `physical_products`

**Test steps**

https://github.com/degica/komoju-woocommerce/assets/15086337/ca0b63ab-cbe7-49da-9442-81bc8fb58360

- Add a product to your cart and go to the checkout page.
- Fill in your billing info and select Paidy at the bottom.
- Proceed with the payment. You should be redirected to the hosted page.
- Proceed with the payment. A Paidy screen should pop up asking for information. This is good enough, you don't need to complete the payment.
